hs-bindgen/src/HsBindgen/C/Fold/Type.hs

```diff
@@ -312,17 +299,16 @@ processTypeDecl' path extBindings unit declCursor ty = case fromSimpleEnum $ cxt
         if anon
         then do
             -- anonymous declaration, nothing to do
-
-            -- TODO: check with struct foo { struct { ... } field; };
+            -- TODO: This is wrong, they can be nested.
```

Comment:
what's wrong?

Reply:
Perhaps I misunderstood what was happening here, but I thought examples like this is "what's wrong":

```c
struct exA {
        enum enumA {
                VALA_1,
                VALA_2
        } fieldA1;
};     

struct exB {
        enum {
                VALB_1,
                VALB_2
        } fieldB1;
};      
```